kratos/utilities/math_utils.h

```diff
@@ -887,11 +887,12 @@ class MathUtils
     
     template< class T1, class T2 , class T3>
     static inline void CrossProduct(T1& c, const T2& a, const T3& b ){
+        if (c.size() != 3) c.resize(3);
 #ifdef KRATOS_DEBUG
-        if (a.size() != 3 || b.size() != 3) 
-        {
-            KRATOS_ERROR << "The size of the vectors is different of 3: " << a << " and " << b << std::endl;
-        }
+        KRATOS_ERROR_IF(a.size() != 3 || b.size() != 3 || c.size() != 3) << "The size of the vectors is different of 3: " << a << ", " << b << " and " << c << std::endl;
+//         // If the direction is the same we have aliasing
+//         KRATOS_ERROR_IF(&c == &a) << "Aliasing between the output parameter and the first input parameter" << std::endl;
```

Comment:
i think you must make a cast, for example for std::size_t for the check to work. for example

KRATOS_ERROR_IF((std::size_t)&c == (std::size_t)&a)

(or even better with a static cast of the pointer direction to std::size_t

Reply:
Yes, I had problems because of the type, I was going to ask waht to do